resources/assets/js/pages/home.js

```diff
@@ -1,14 +1,74 @@
 (function() {
     var showDialogButton = document.querySelector('#fire'),
-        formDeploy = document.querySelector('form');
+        formDeploy = document.querySelector('form'),
+        serverSelect = document.querySelector('#select-server'),
+        branchSelect = document.querySelector('#select-branch'),
+        taskSelect = document.querySelector('#select-task'),
+        progressBar = document.querySelector('#progress-bar');
 
     if (showDialogButton) {
         var dialog = new DeployerConfirmDialog();
 
-        showDialogButton.addEventListener('click', function() {
+        showDialogButton.addEventListener('click', function(event) {
+            event.preventDefault();
+
+            if(serverSelect.value === "-1") {
```

Comment:
also, what about create an array and loop into these vars `serverSelect, ...` doing check as it's the very same logic